--- src/build/config/allocator.gni.orig
+++ src/build/config/allocator.gni
@@ -16,7 +16,7 @@
 # The debug CRT on Windows has some debug features that are incompatible with
 # the shim. NaCl in particular does seem to link some binaries statically
 # against the debug CRT with "is_nacl=false".
-if ((is_linux || is_android || is_mac ||
+if (((is_linux && is_gnu_linux) || is_android || is_mac ||
      (is_win && !is_component_build && !is_debug)) && !is_asan && !is_lsan &&
     !is_tsan && !is_msan) {
   _default_use_allocator_shim = true
