--- electron.gyp.orig
+++ electron.gyp
@@ -262,6 +262,8 @@
         '<(libchromiumcontent_src_dir)/v8/include',
         # The `node.h` is using `#include"ares.h"`.
         'vendor/node/deps/cares/include',
+        '<(source_root)/vendor/node/deps/cares/include',
+        '<(source_root)/vendor/node/deps/http_parser',
         # The `third_party/WebKit/Source/platform/weborigin/SchemeRegistry.h` is using `platform/PlatformExport.h`.
         '<(libchromiumcontent_src_dir)/third_party/WebKit/Source',
         # The 'third_party/libyuv/include/libyuv/scale_argb.h' is using 'libyuv/basic_types.h'.
--- vendor/node/node.gyp.orig
+++ vendor/node/node.gyp
@@ -157,6 +157,8 @@
         'tools/msvs/genfiles',
         'deps/uv/src/ares',
         '<(SHARED_INTERMEDIATE_DIR)', # for node_natives.h
+        'deps/cares/include',
+        'deps/http_parser'
       ],
 
       'sources': [
